felix/wireguard/wireguard.go

```diff
@@ -1496,6 +1498,17 @@ func (w *Wireguard) ensureLink(netlinkClient netlinkshim.Interface) (bool, error
 		}
 	}
 
+	// Can only enable NAPI threading once the link is up
+	if attrs.Flags&net.FlagUp != 0 {
+		threadedNAPIBit := boolToBinaryString(w.config.ThreadedNAPI)
+		w.logCtx.WithField("flags", attrs.Flags).Info(fmt.Sprintf("Set NAPI threading to %s for wireguard interface %s", threadedNAPIBit, w.interfaceName))
+		napiThreadedPath := fmt.Sprintf("/sys/class/net/%s/threaded", w.interfaceName)
+		if err := w.writeProcSys(napiThreadedPath, threadedNAPIBit); err != nil {
+			w.logCtx.WithError(err).Warn(fmt.Sprintf("failed to set NAPI threading to %s for wireguard for interface %s", threadedNAPIBit, w.interfaceName))
+			return false, err
```

Comment:
Do we need to fail here? IIUC, not enabling NAPI will leads to packet drop under high pps. I think it's probably better warn user and continue as usual. WDYT?
@coutinhop

Reply:
We can just warn about the setting without failing. That way `ensureLink` and wireguard is successful but it's clear in the logs that NAPI wasn't toggled.